rope/base/codeanalyze.py

```diff
@@ -181,7 +181,7 @@ def logical_line_in(self, line_number):
                 indents = count_line_indents(self.lines.get_line(lineno))
 
     def generate_starts(self, start_line=1, end_line=None):
-        for start, end in self.generate_regions(start_line, end_line):
+        for start, _end in self.generate_regions(start_line, end_line):
```

Comment:
Having a convention where unused variable have to be prefixed with underscore (e.g. `_varname`) isn't going to be enforceable unless we run a linter scanning for this as pre-commit or as a Github Actions; it's too easy to miss that during manual code review. I'd rather we either disable this warning in local linter or add at least a Github Actions to enforce this.

Reply:
flake8 was the source of the complaints. Is it hard to check commits with flake8?

Similarly, it might be good to add a GitHub action that reindent does nothing on all files.

Reply:
`B007` is is now disabled. We can revisit this question at our leisure.

Reply:
> reindent does nothing on all files, including those in the ropetest directory.

We have Black running on GitHub Actions for formatting checks. I think that should be sufficient, adding more than one formatter would be undesirable, IMO, as it creates the possibility of conflicting advices from the tools.

Reply:
Good to know.